src/preprocessing/passes/non_clausal_simp.h

```diff
@@ -48,6 +49,15 @@ class NonClausalSimp : public PreprocessingPass
 
   /** Learned literals */
   std::vector<Node> d_nonClausalLearnedLiterals;
+
+  /** Preprocess the boolean skeleton,
+  which calls solveByCryptominisat or solveByCircuitPropagator */
+  std::pair<bool, std::vector<Node>> preprocessByCryptominisat(
+      AssertionPipeline* assertionsToPreprocess, unsigned substs_index);
+
+  std::pair<bool, std::vector<Node>> preprocessByCircuitPropagator(
```

Comment:
Add a comment for this method (also describe the arguments it expects and what it returns).